node_trans_session.js

```diff
@@ -88,9 +88,12 @@ class NodeTransSession extends EventEmitter {
               || filename.endsWith('.tmp')) {
               fs.unlinkSync(ouPath + '/' + filename);
             }
-          })
+          });
         }
       });
+      if (this.conf.hls) {
+        fs.writeFileSync(ouPath + '/' + this.hlsFileName, '#EXTM3U\n');
+      }
```

Comment:
This logic can be determined at the point of deleting the file, instead of deleting and creating it again. 
Also, there is a possibility that the file still exist because of the `if(!err)` check, and you still attempt to create it anyways.

Would it be better to move these to the `readdirSync` callback?